src/runtime/components/Form.vue

```diff
@@ -189,17 +205,30 @@ async function _validate<T extends boolean>(opts: ValidateOpts<true, T>): Promis
 async function _validate<T extends boolean>(opts: ValidateOpts<boolean, boolean> = { silent: false, nested: true, transform: false }): Promise<FormData<S, T> | false> {
   const names = opts.name && !Array.isArray(opts.name) ? [opts.name] : opts.name as (keyof O)[]
 
+  async function validateNestedForms({ validate, name }: { validate: typeof _validate, name?: string }) {
+    try {
+      return { name, output: await validate({ ...opts, silent: false }) }
+    } catch (error: unknown) {
+      if (!(error instanceof FormValidationException)) {
+        throw error
+      }
+      return { name, error }
+    }
+  }
+
   const nestedValidatePromises = !names && opts.nested
-    ? Array.from(nestedForms.value.values()).map(
-        ({ validate }) => validate(opts as any).then(() => undefined).catch((error: Error) => {
-          if (!(error instanceof FormValidationException)) {
-            throw error
-          }
-          return error
-        })
-      )
+    ? Array.from(nestedForms.value.values()).map(validateNestedForms)
     : []
 
+  const nestedResults = await Promise.all(nestedValidatePromises)
+  const nestedErrors = nestedResults.flatMap((c) => {
```

Comment:
We gather errors from nested forms, but errors from the parent are not injected in nested forms. This means that the exposed `getErrors` method will return all errors, but calling `setErrors` or `clear` on the parent won't affect it's children.